src/worker.ts

```diff
@@ -0,0 +1,75 @@
+import { Value } from "@sinclair/typebox/value";
+import { run } from "./run";
+import { Env, envConfigValidator } from "./types/env";
+import { assistivePricingSettingsSchema } from "./types/plugin-input";
+
+export default {
+  async fetch(request: Request, env: Env): Promise<Response> {
+    try {
+      if (request.method !== "POST") {
+        return new Response(JSON.stringify({ error: `Only POST requests are supported.` }), {
+          status: 405,
+          headers: { "content-type": "application/json", Allow: "POST" },
+        });
+      }
+      const contentType = request.headers.get("content-type");
+      if (contentType !== "application/json") {
+        return new Response(JSON.stringify({ error: `Error: ${contentType} is not a valid content type` }), {
+          status: 400,
```

Comment:
5xx are errors

Reply:
`400` means invalid arguments, which is the case here. I'll replace `Error` with `Bad Request`.